crates/shared/src/order_validation.rs

```diff
@@ -787,11 +803,18 @@ pub async fn get_quote_and_check_fee(
     quote_search_parameters: &QuoteSearchParameters,
     quote_id: Option<i64>,
     fee_amount: Option<U256>,
+    market_orders_deprecation_date: Option<chrono::DateTime<chrono::Utc>>,
```

Comment:
nit: Technically we could have a single Option around the fee amount and market order deprecation date, which would be a bit cleaner (but since we are removing the deprecation date soon it's not a big deal)